### PR TITLE
Fix Swift breaks by correcting issue with Gem adding EarlGrey.swift irrespective of the type of the project.

### PR DESCRIPTION
**Pull request by tirodkar**
Also deletes unneeded Swift App imports in the project which was causing the ObjC tests to also contain EarlGrey.swift.

### CODE REVIEW

File: gem/lib/earlgrey/configure_earlgrey.rb [hunk 1]
```diff
@@ -353,7 +354,21 @@ def add_earlgrey_framework(target, framework_ref)
...
       target.source_build_phase.files_references.any? do |ref|
-        SWIFT_FILETYPE == (ref.last_known_file_type || ref.explicit_file_type)
+        file_type = ref.last_known_file_type
+        if file_type.to_s.empty?
+          file_type = ref.explicit_file_type
+        end
+
+        if SWIFT_FILETYPE == file_type
+          return true
+        elsif OBJC_FILETYPE == file_type
```
Comment (wuhao5) [inline gem/lib/earlgrey/configure_earlgrey.rb]: why would you compare objc? it can have both objc and swift source files, not?

File: Demo/EarlGreyExample/EarlGreyExample.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -532,6 +530,7 @@
 				SWIFT_OBJC_BRIDGING_HEADER = "";
 				SWIFT_OBJC_INTERFACE_HEADER_NAME = "EarlGreyExampleSwift-Swift.h";
 				SWIFT_OPTIMIZATION_LEVEL = "-Onone";
+				SWIFT_VERSION = "";
```
Comment (wuhao5) [inline Demo/EarlGreyExample/EarlGreyExample.xcodeproj/project.pbxproj]: revert this?

File: Demo/EarlGreyExample/EarlGreyExample.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -562,6 +561,7 @@
 				PRODUCT_NAME = "$(TARGET_NAME)";
 				SWIFT_OBJC_BRIDGING_HEADER = "";
 				SWIFT_OBJC_INTERFACE_HEADER_NAME = "EarlGreyExampleSwift-Swift.h";
+				SWIFT_VERSION = "";
```
Comment (wuhao5) [inline Demo/EarlGreyExample/EarlGreyExample.xcodeproj/project.pbxproj]: revert this as well

Comment (wuhao5) [inline Demo/EarlGreyExample/EarlGreyExample.xcodeproj/project.pbxproj]: i think you should remove the default value from the project level. unspecified can also indicate it has swift sources, right?

File: gem/lib/earlgrey/configure_earlgrey.rb [hunk 4]
```diff
@@ -364,8 +379,7 @@ def has_swift?(target)
     # @param [Xcodeproj::Project] project
     # @param [PBXNativeTarget] target
     def copy_swift_files(project, target, swift_version = nil)
-      return unless has_swift?(target) || !swift_version.nil?
-
+      return unless has_swift?(target) || !swift_version.to_s.empty?
```
Comment (wuhao5) [inline gem/lib/earlgrey/configure_earlgrey.rb]: I don't know the semantic meaning of Unspecified or empty.

my guess is if you have swift files but doesn't specify its version, it would throw an error when compiling?

File: Scripts/travis.sh [hunk 5]
```diff
@@ -108,8 +108,9 @@ elif [ "${TYPE}" == "EXAMPLE_PODS_SWIFT" ]; then
   cd gem/
   bundle install
   rake install
-  cd ..
-  pod install --project-directory=Demo/EarlGreyExample
```
Comment (wuhao5) [inline Scripts/travis.sh]: does it not work any more?

File: gem/lib/earlgrey/configure_earlgrey.rb (hunk 1 above)
Comment (wuhao5) [inline gem/lib/earlgrey/configure_earlgrey.rb]: I think this actually doesn't make sense if you look at the method name. you can have a separate method like has_objc?, but I still don't know if you really need to check that.

also throwing the error doesn't seem to be a good idea here, has_swift? returns false if it doesnt have swift files.

Comment (tirodkar) [inline gem/lib/earlgrey/configure_earlgrey.rb]: The method still checks if Swift is present. Just ensures that something is explicitly set.

You can have ObjC and Swift in the same target as we did with EGExampletests. But it came up as file type - Swift. I don't think both can be nil.

File: Demo/EarlGreyExample/EarlGreyExample.xcodeproj/project.pbxproj (hunk 3 above)
Comment (tirodkar) [inline Demo/EarlGreyExample/EarlGreyExample.xcodeproj/project.pbxproj]: On changing the value at the project level, I received the following error:

```
“Swift Language Version” (SWIFT_VERSION) is required to be configured correctly for targets which use Swift. Use the [Edit > Convert > To Current Swift Syntax…] menu to choose a Swift version or use the Build Settings editor to configure the build setting directly.
```

File: Scripts/travis.sh (hunk 5 above)
Comment (tirodkar) [inline Scripts/travis.sh]: Moved this to a separate CL. Reverted.

File: gem/lib/earlgrey/configure_earlgrey.rb (hunk 4 above)
Comment (tirodkar) [inline gem/lib/earlgrey/configure_earlgrey.rb]: That's correct. Hmmm... should change this to an & then?

File: gem/lib/earlgrey/configure_earlgrey.rb (hunk 1 above)
Comment (wuhao5) [inline gem/lib/earlgrey/configure_earlgrey.rb]: The logic I am reading here:
go thru each file type, if one objc encounters, return false; if one swift file encounters, return true.

so, let's say you have: [a.m, b.swift], it will return false, where it should return true.

I don't quite follow the reason behind...

File: Demo/EarlGreyExample/EarlGreyExample.xcodeproj/project.pbxproj (hunk 3 above)
Comment (wuhao5) [inline Demo/EarlGreyExample/EarlGreyExample.xcodeproj/project.pbxproj]: I see, I thought this target doesn't use swift.

File: gem/lib/earlgrey/configure_earlgrey.rb (hunk 1 above)
Comment (tirodkar) [inline gem/lib/earlgrey/configure_earlgrey.rb]: I just re-tested this out. This method does indeed seem to be working correctly.

The issue was the version number, even though it was set to zero, was being returned as true, which led to the Swift bundling being done regardless. I'll change this back!

Comment (tirodkar) [inline gem/lib/earlgrey/configure_earlgrey.rb]: Done.

File: Demo/EarlGreyExample/EarlGreyExample.xcodeproj/project.pbxproj (hunk 3 above)
Comment (tirodkar) [inline Demo/EarlGreyExample/EarlGreyExample.xcodeproj/project.pbxproj]: It used to have a Swift file, which is why it was set to 3.0. Without this, it'll inherit from the parent project which has 3.0, which will trigger the Swift File Bundling in configure_earlgrey..., which will cause the following error:

```
RuntimeError - Bundled swift missing
/Users/tirodkar/.rvm/gems/ruby-2.3.1/gems/earlgrey-0.2.0/lib/earlgrey/configure_earlgrey.rb:382:in `copy_swift_files'
```

File: gem/lib/earlgrey/configure_earlgrey.rb [hunk 6]
```diff
@@ -41,6 +41,7 @@ module EarlGrey
   HEADER_SEARCH_PATHS = 'HEADER_SEARCH_PATHS'.freeze
 
   SWIFT_FILETYPE = 'sourcecode.swift'.freeze
+  OBJC_FILETYPE = 'sourcecode.c.objc'.freeze
```
Comment (wuhao5) [inline gem/lib/earlgrey/configure_earlgrey.rb]: revert this.

Comment (tirodkar) [inline gem/lib/earlgrey/configure_earlgrey.rb]: Done
